### DISCUSSION TITLE
Properly get `defaultRadius` from the theme

### DISCUSSION THREAD

**Discussion by BenJeau** [mantinedev/mantine#1800] [Q&A]
Hi, thanks for such a great library!

I'm changing the style of one of the components, and I'd like to access the `defaultRadius` from the theme, it seems like I need to do the following:

```ts
theme.radius[theme.defaultRadius]
```

But then typescript complains with:

```sh
Element implicitly has an 'any' type because expression of type '(string & {}) | MantineNumberSize' can't be used to index type 'MantineSizes'.
  No index signature with a parameter of type 'number' was found on type 'MantineSizes'
```

I understand the error, since `theme.defaultRadius` could be a string or number. I was wondering what is the way you'd suggest to access the default radius?

Thanks 😄  (I don't love having a bunch of `// @ts-ignore`)

**Comment by rtivital** [comment]
Use `theme.fn.radius()`

**Reply by BenJeau** [reply]
Thanks that works! But I got another TS error, 

```sh
(method) MantineThemeFunctions.radius(size: (string & {}) | MantineNumberSize): string | number
Expected 1 arguments, but got 0.
MantineTheme.d.ts(35, 12): An argument for 'size' was not provided.
```

But looking at the code itself, the type for that function should rather be `MantineNumberSize | (string & {}) | undefined`? If you don't mind, I'll create a small PR with that change

**Reply by rtivital** [reply]
Try using `theme.fn.radius(theme.defaultRadius)`, current definition must be correct

**Reply by BenJeau** [reply]
That works, but looking at the code, if its null/undefined, it will use the `theme.defaultRadius` - there even seems to be a test case for this case

https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/functions/fns/radius/radius.test.ts#L13-L21

So maybe it the param should be optional? I've created a PR if you think the change should be made https://github.com/mantinedev/mantine/pull/1803
